pkg/statistics/handle/cache/statscache.go

```diff
@@ -65,6 +65,47 @@ func NewStatsCacheImplForTest() (types.StatsCache, error) {
 	return NewStatsCacheImpl(nil)
 }
 
+type tablesToUpdateOrDelete struct {
+	toUpdate []*statistics.Table
+	toDelete []int64
+	option   types.UpdateOptions
+}
+
+func (t *tablesToUpdateOrDelete) addToUpdate(statsCache *StatsCacheImpl, table *statistics.Table) {
+	if len(t.toUpdate) == 10 {
+		statsCache.UpdateStatsCache(types.CacheUpdate{
+			Updated: t.toUpdate,
+			Deleted: t.toDelete,
+			Options: t.option,
+		})
+		t.toUpdate = t.toUpdate[:0]
+		t.toDelete = t.toDelete[:0]
+	}
+	t.toUpdate = append(t.toUpdate, table)
+}
+
+func (t *tablesToUpdateOrDelete) addToDelete(statsCache *StatsCacheImpl, tableID int64) {
+	if len(t.toDelete) == 10 {
+		statsCache.UpdateStatsCache(types.CacheUpdate{
+			Updated: t.toUpdate,
+			Deleted: t.toDelete,
+			Options: t.option,
+		})
+		t.toUpdate = t.toUpdate[:0]
+		t.toDelete = t.toDelete[:0]
```

Comment:
clear(t.ToDelete)

Reply:
The clear seems to be reset to zero instead of reset the length.
```
// The clear built-in function clears maps and slices.
// For maps, clear deletes all entries, resulting in an empty map.
// For slices, clear sets all elements up to the length of the slice
// to the zero value of the respective element type. If the argument
// type is a type parameter, the type parameter's type set must
// contain only map or slice types, and clear performs the operation
// implied by the type argument.
```